cupyx/scipy/signal/_peak_finding.py

```diff
@@ -761,14 +762,28 @@ def _arg_peaks_as_expected(value):
     return value
 
 
-@jit.rawkernel()
-def _check_prominence_invalid(n, peaks, left_bases, right_bases, out):
-    tid = jit.blockIdx.x * jit.blockDim.x + jit.threadIdx.x
-    i_min = left_bases[tid]
-    i_max = right_bases[tid]
-    peak = peaks[tid]
-    valid = 0 <= i_min and i_min <= peak and peak <= i_max and i_max < n
-    out[tid] = not valid
+_check_prominence_invalid = cupy.RawKernel(r"""
+extern "C" __global__ void check_prominence_invalid
+(
+int n,
+int x_n,
+long long* peaks,
+long long* left_bases,
+long long* right_bases,
+bool* out
+) {
+    int idx = blockDim.x * blockIdx.x + threadIdx.x;
+    if(idx >= n) {
+        return;
+    }
+
+    long long i_min = left_bases[idx];
+    long long i_max = right_bases[idx];
+    long long peak = peaks[idx];
+    bool valid = 0 <= i_min && i_min <= peak && peak <= i_max && i_max < x_n;
+    out[idx] = !valid;
+}
+""", 'check_prominence_invalid')
```

Comment:
With this sort of thing, wonder if it is worth putting the CUDA code into a sidecar `.cu` file and simply reading it in here

The advantage being that normal tools (editors, linters, compilers, etc.) for working with a CUDA file are usable

The main disadvantage seems to be indirection. Though this might be worth it for the advantages above

Reply:
Throughout the CZI grant we have been implementing kernels inline as strings, as opposed to including more compiled dependencies, since that would increase build times

Reply:
To be clear the suggestion wasn't to compile it. Just to put this text in another file and read from it

If the concern is the IO of reading would be slow, we could consider caching the result

Reply:
For instance, https://github.com/cupy/cupy/blob/main/cupyx/scipy/signal/_iir_utils.py  and https://github.com/cupy/cupy/blob/main/cupyx/scipy/spatial/delaunay_2d/_kernels.py are full of kernel definitions. At some point, we tried to add some compiled code, and this was the official reply from @asi1024: https://github.com/cupy/cupy/pull/7985#issuecomment-1867322678

Reply:
I don't know if it would be worth to detail a separate issue to convert the inline strings into their own `.cu` files? AFAIK, all of the new signal and interpolate implementations are inline strings

Reply:
That's fair. Maybe this should be its own issue

Mostly noting this as an observation I've had maintaining CuPy code using this inline string pattern and working with others doing the same. Think we may be better served maintenance-wise putting the CUDA code in a `.cu` file and then loading it here

Just makes it easier to edit, lint, catch bugs, etc.

Reply:
Filed issue: https://github.com/cupy/cupy/issues/8581

We can leave as-is here and discuss there